OSDOCS-2933 Java OSDK tutorial (#47560)
openshift/openshift-docs: modules/osdk-java-generate-crd.adoc

// Module included in the following assemblies:
//
// * operators/operator_sdk/java/osdk-java-tutorial.adoc

:_content-type: PROCEDURE
[id="osdk-java-generate-crd_{context}"]
= Generating CRD manifests

After the API is defined with `MemcachedSpec` and `MemcachedStatus` files, you can generate CRD manifests.

.Procedure

> * Run the following command from the previously created `memcached-operator` directory to generate the CRD:

Reviewer: You could also add an extra step: Run the following command to change into the previously created `memcached-operator` directory: `$ cd memcached-operator`.

+
[source,terminal]
----
$ mvn clean install
----

.Verification

> * The CRD is stored in the `/target/kubernetes` directory. The contents of the `memcacheds.cache.example.com-v1.yml` file is shown below:

Reviewer 2: Suggestion to make this more of a verification step (vs. like an output):
```suggestion
* The CRD is stored in the `/target/kubernetes` directory. Run the following to examine the contents of the `memcacheds.cache.example.com-v1.yml` file, as shown below:
+
[source,terminal]
----
$ cat memcacheds.cache.example.com-v1.yaml
----
+
.Example output
```

Reviewer: I second the `.Example` label, and would go a step further to include the file and filepath. So `.Example target/kubernetes/memcacheds.cache.example.com-v1.yml` file

It is a mouthful, but I think it helps with scanability in these long documents where you are editing a lot of files.

Reviewer: You could always truncate the filepath if it renders weird.

> * The CRD is stored in the `/target/kubernetes` directory. The contents of the `memcacheds.cache.example.com-v1.yml` file is shown below:

Reviewer: ```suggestion
* Verify the contents of the CRD in the `target/kubernetes/memcacheds.cache.example.com-v1.yml` file as shown in the following example:
```
Suggestion to remove passive voice and to avoid [_below_](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#b).

Reviewer 3: +1 to this suggestion
